utils/json-utils/src/main/scala/pl/touk/nussknacker/engine/json/swagger/extractor/JsonToNuStruct.scala

```diff
@@ -26,13 +26,17 @@ object JsonToNuStruct {
 
     def addPath(next: String): String = if (path.isEmpty) next else s"$path.$next"
 
-    def extractObject(elementType: Map[PropertyName, SwaggerTyped]): AnyRef =
+    def extractObject(obj: SwaggerObject): AnyRef =
       extract[JsonObject](
         _.asObject,
         jo => TypedMap(
           jo.toMap.collect {
-            case (key, value) if elementType.contains(key) =>
-              key -> JsonToNuStruct(value, elementType(key), addPath(key))
+            case (key, value) if obj.elementType.contains(key) =>
+              key -> JsonToNuStruct(value, obj.elementType(key), addPath(key))
+            case (key, value) if obj.additionalProperties.isInstanceOf[AdditionalPropertiesSwaggerTyped] =>
+              key -> JsonToNuStruct(value, obj.additionalProperties.asInstanceOf[AdditionalPropertiesSwaggerTyped].value, addPath(key))
+            case (key, value) if obj.additionalProperties.asInstanceOf[AdditionalPropertiesBoolean].value =>
+              key -> jsonToAny(value)
```

Comment:
do you handle here situation when `addionalProperties == false`?
and also can we get rid of `isInstanceOf` nad `asInstanceOf` since we have seald trait?

Reply:
```
  test("should trim useless fields") {
    val definition = SwaggerObject(elementType = Map("field3" -> SwaggerLong), AdditionalPropertiesDisabled)
    extractor.JsonToNuStruct(json, definition) shouldBe TypedMap(Map.empty)
  }``